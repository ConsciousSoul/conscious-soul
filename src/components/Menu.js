import React, { useEffect, useState } from 'react';
import {MenuWrapper, Button} from '../styles/menu/MenuStyles.css'
import { fetchData } from '../requests.js'

export const Menu = (props) => {
  const [menuItems, setMenuItems] = useState([])

  useEffect(() => {
    fetchData('/api/v1/menu_items')
      .then(res => setMenuItems(res.data))
      .catch(error => console.log(error))
  })

  return (
    <MenuWrapper>
        <Button onClick={props.closeMenu}>Return to Main Page</Button>
        <p>{JSON.stringify(menuItems)}</p>
      </MenuWrapper>
  )
}

export default Menu;
