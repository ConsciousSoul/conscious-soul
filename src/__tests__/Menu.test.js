import React from 'react';
import { shallow } from 'enzyme';
import Menu from '../components/Menu';
import {MenuWrapper, Button} from '../styles/menu/MenuStyles.css.js';

describe('Menu component', () => {
  describe('renders', () => {
    it('should not crash', () => {
      shallow(<Menu />);
    });

    it('should have a return button', () => {
      const wrapper = shallow(<Menu />);
      expect(wrapper.find('Button').text()).toEqual('Return to Main Page');
    });

    it('should have the correct styles', () => {
      expect(shallow(<MenuWrapper />)).toMatchSnapshot();
      expect(shallow(<Button />)).toMatchSnapshot();
    });
  });
});