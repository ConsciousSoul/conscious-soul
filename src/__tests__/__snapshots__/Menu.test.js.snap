// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders should have the correct styles 1`] = `ShallowWrapper {}`;

exports[`Menu component renders should have the correct styles 2`] = `ShallowWrapper {}`;
